Code/Tools/AssetProcessor/native/resourcecompiler/rcjob.cpp

```diff
@@ -876,6 +898,20 @@ namespace AssetProcessor
             return false;
         }
 
+        auto* uuidInterface = AZ::Interface<AzToolsFramework::IUuidUtil>::Get();
+
+        if (!uuidInterface)
+        {
+            AZ_Assert(false, "Programmer Error - IUuidUtil interface is not available");
+            return false;
+        }
+
+        // Go through all the intermediate products and output the assigned UUID
+        for (auto [intermediateProduct, uuid] : intermediateOutputPaths)
+        {
+            uuidInterface->CreateSourceUuid(intermediateProduct.toUtf8().constData(), uuid);
```

Comment:
Nit: May worth checking the return value here for potential errors?

Reply:
Added